### DISCUSSION TITLE
Help with Docker build

### DISCUSSION THREAD

**Discussion by alidaf** [sle118/squeezelite-esp32#284] [Q&A] [answered]
I'm running the Docker image and this command fails with CMakeLists.txt not found in project directory.

I'm running the Docker image on a Mac using `docker run -it -v `pwd`:/workspace/squeezelite-esp32 sle118/squeezelite-esp32-idfv435`

**Comment by sle118** [comment]
I am assuming that you've cloned the project locally before attempting to build it? I will move this to the Q&A discussion section. 

Out of curiosity, what is your motivation to build the project instead of using one of the stable builds already available?

**Comment by alidaf** [comment]
Yes, I cloned the project. I managed to sort out that issue (I had to be in the proper folder, which isn't explained in the guide) but then ran into errors to do with Spotify so I'm now trying to use the Espressif installation, which is a headache - lot's of missing dependencies.

To answer your question, I designed a case that I 3D printed (shown below) - it generated a lot of interest and I received a request to consider ESP32 (I currently use RPi with piCorePlayer). 

![DSC_1602](https://github.com/sle118/squeezelite-esp32/assets/14184640/6b35d0e7-54b0-4846-8167-a9c635dc5984)

It seems to be a better option (certainly cheaper), but I'm not familiar with the platform. I asked around for supply of the SqueezeAMP board but got no responses so I bought 2 WROVER dev boards (ESP32-DevkitC) to get to grips with it. I don't need some of the SqueezeAMP features - I only want Analogue Out or SPDIF, A display and rotary encoder, so the amplifier is a waste for me. I'm more familiar with the Arduino platform and IDE so this is different enough to be a challenge. I'm trying to build it piece-by-piece, starting with the display (ST7789V). 

The case is going to be part of a series - Squeeze, Amp, visualiser, power supply.

**Comment by alidaf** [comment]
Here's another image.

![DSC_1600](https://github.com/sle118/squeezelite-esp32/assets/14184640/4f019e06-76d7-4c80-a2a2-7902b1c804d0)

Currently there is only USB out because my AMP has a DAC module that has multiple SPDIF inputs as well as USB.

![DSC_1598](https://github.com/sle118/squeezelite-esp32/assets/14184640/26af44b1-aa9d-4226-9dfd-f22fabc36b38)

But, demand will probably dictate that I need something with analogue out.

**Comment by alidaf** [comment]
It seems like it's no longer possible to install esp32 in the v4.3.5 tree.

**Comment by sle118** [comment]
This is a very nice project you have going, and a neat case design. Typically, though, there should be no need to build your own binaries, as the i2s-4mb target is generic enough to be used on pretty much any device running a WROVER. 

I am curious to know how you tried installing the firmware. I haven't tried our web installer in a while, but it is visited every week by a good number of users and I have not heard of it failing.

@philippe44 took some distance from the project for a few weeks and SqueezeAmp being one of his hobby, there wasn't any plan for a new batch until a couple of weeks back where he announced a possible remake in the Logitech forums. I like it's design but coming up with your own to fulfill your requirements is definitely more gratifying when you gave the knowledge and time to do it.

I'll try to help you with that.

**Reply by alidaf** [reply]
Ah, I thought that the display and other peripheral gpios needed configuring in menuconfig.

How is this done otherwise then? I've been trying to compile with the Docker and Esspressif installations but both have problems. I haven't tried to upload an image yet - I have no idea how that's going to work on a Mac. I've been wondering whether to use a RPi4 to build and upload. I'd hoped that I could use the Arduino IDE to build since I'm more familiar with it but I don't think it will be possible without a lot of additional configuration.

**Reply by alidaf** [reply]
Also, thanks for the comment. 
I visited Waterloo University many years ago - it's in Quebec, right?. It is one of the most beautiful places I've ever been. I loved Toronto too.

**Reply by sle118** [reply]
Waterloo University is in Ontario. Not too far from Quebec, all things considered and depending where you came from. Toronto to Montreal is a seriously long drive 😃

**Comment by alidaf** [comment]
@philippe44 I have managed to install the generic firmware via the web installer, have it connected to my network and can use the configuration page. 
The SPI settings are confusing though.
I assume Data OUT is MISO and Data IN is MOSI. When I enter the values, it reboots and Data IN is blank. I have filled out the Display settings but still nothing. I have tried swapping IN and OUT too.

**Comment by alidaf** [comment]
Well, I looked in the NVS editor and my assumption was incorrect. I changed the settings again and now have an IP address on the display. I need to rotate it but the NVS tab has disappeared.

**Comment by alidaf** [comment]
Credits tab has a show NVS!

**Reply by philippe44** [reply, marked as answer]
If you enter nvs parameters manually, the README should give you all explanations you need. Otherwise, @sle118 has developed the UI that should make adding a display much easier. About recompiling, the esp-idf should be pretty straightforward. Clone it, checkout the version release-v43, init submodule with --recursive and then do the install. You also have to install using pip3 protobuf and grpcio-tool (see cspot on @feelfreelinux) for exact syntax. After it will build immediately. I just did that on a new computer yesterday

**Reply by sle118** [reply]
@alidaf definitely use the UI for configuration if you are new to the nvs parameters of our system, as it will make your life easier.
